CRM/Core/DAO.php

```diff
@@ -962,7 +962,9 @@ public static function deleteRecord(array $record) {
     CRM_Utils_Hook::pre('delete', $entityName, $record['id'], $record);
     $instance = new $className();
     $instance->id = $record['id'];
-    if (!$instance->delete()) {
+    // Load complete object for the sake of post-hooks
+    $instance->find(TRUE);
```

Comment:
@colemanw could you add a comment here explaining what you explained on chat - ie that we can't expect post hooks to do this load themselve in the case of delete as there is nothing left to load. With that added this is MOP